demos/http/http_demo_s3_download/http_demo_s3_download.c

```diff
@@ -215,14 +362,423 @@ static bool getS3ObjectFileSize( size_t * pFileSize,
+
+/**
+ * @brief Application-defined Hash Final function provided to the SigV4 library.
+ *
+ * @note Refer to SigV4CryptoInterface_t interface documentation for this function.
+ */
+static int32_t sha256Final( void * hashContext,
+                            char * pOutput,
+                            size_t outputLen );
+
+/**
+ * @brief CryptoInterface provided to SigV4 library for generating the hash digest
+ * and signature.
+ */
+static SigV4CryptoInterface_t cryptoInterface =
+{
+    .hashInit      = sha256Init,
+    .hashUpdate    = sha256Update,
+    .hashFinal     = sha256Final,
+    .pHashContext  = &hashContext,
+    .hashBlockLen  = SHA256_HASH_MAX_BLOCK_LENGTH,
+    .hashDigestLen = SHA256_HASH_MAX_DIGEST_LENGTH,
+};
+
+/**
+ * @brief SigV4 parameters provided to SigV4 library by the application for generating
+ * the Authorization header.
+ */
+static SigV4Parameters_t sigv4Params =
+{
+    .pCredentials     = &sigvCreds,
+    .pDateIso8601     = pDateISO8601,
+    .pRegion          = AWS_S3_BUCKET_REGION,
+    .regionLen        = strlen( AWS_S3_BUCKET_REGION ),
+    .pService         = AWS_SERVICE_NAME,
+    .serviceLen       = strlen( AWS_SERVICE_NAME ),
+    .pCryptoInterface = &cryptoInterface,
+    .pHttpParameters  = NULL
+};
+
+/*-----------------------------------------------------------*/
+
+static bool getTemporaryCredentials( TransportInterface_t * transportInterface,
+                                     size_t pDateISO8601Len,
+                                     HTTPResponse_t * response,
+                                     char * pDateISO8601,
+                                     SigV4Credentials_t * sigvCreds )
+{
+    bool returnStatus = true;
+    HTTPRequestHeaders_t requestHeaders = { 0 };
+    HTTPRequestInfo_t requestInfo = { 0 };
+    size_t pathLen = 0;
+    size_t addressLen = 0;
+    HTTPStatus_t httpStatus = HTTPSuccess;
+    SigV4Status_t sigv4Status = SigV4Success;
+    JSONStatus_t jsonStatus = JSONSuccess;
+    const char * pAddress = NULL;
+    const char * pDate;
+    const char * pPath;
+    size_t dateLen;
+
+    assert( transportInterface != NULL );
+    assert( response != NULL );
+    assert( sigvCreds != NULL );
+    assert( pDateISO8601 != NULL );
+    assert( pDateISO8601Len > 0 );
+
+    /* Retrieve the address location and length from AWS_IOT_CREDENTIAL_PROVIDER_FULL_ENDPOINT. */
+    if( returnStatus == true )
+    {
+        httpStatus = getUrlAddress( AWS_IOT_CREDENTIAL_PROVIDER_FULL_ENDPOINT,
+                                    sizeof( AWS_IOT_CREDENTIAL_PROVIDER_FULL_ENDPOINT ) - 1,
+                                    &pAddress,
+                                    &addressLen );
+
+        if( httpStatus != HTTPSuccess )
+        {
+            LogError( ( "Failed to get Address from URL %s: Error=%s.",
+                        AWS_IOT_CREDENTIAL_PROVIDER_FULL_ENDPOINT, HTTPClient_strerror( httpStatus ) ) );
+            returnStatus = false;
```

Comment:
Does it make sense to use parse out the endpoint and URI path information from a string that is statically constructed in the `AWS_IOT_CREDENTIAL_PROVIDER_FULL_ENDPOINT` logic  as
```
#define AWS_IOT_CREDENTIAL_PROVIDER_FULL_ENDPOINT         \
    "https://"                                            \
    AWS_IOT_CREDENTIAL_PROVIDER_ENDPOINT "/role-aliases/" \
    AWS_IOT_CREDENTIAL_PROVIDER_ROLE "/credentials"
```
It seems like the both the URI path and the endpoint can be stored as separate macro strings instead of having them to be extracted at runtime